src/serial.rs

```diff
@@ -586,6 +594,31 @@ mod tests {
         fn tx_lost_byte(&self) {
             self.tx_lost_byte_count.inc();
         }
+
+        fn in_buffer_empty(&self) {
+            match self
+                .buffer_ready_event
+                .as_ref()
+                .map_or(Ok(()), |buf_ready| buf_ready.write(1))
+            {
+                Ok(_) => (),
+                Err(err) => println!(
+                    "Could not signal that serial device buffer is ready: {:?}",
+                    err
+                ),
+            }
+        }
+    }
+
+    impl Default for ExampleSerialMetrics {
+        fn default() -> Self {
+            ExampleSerialMetrics {
+                read_count: AtomicU64::new(0),
+                out_byte_count: AtomicU64::new(0),
+                tx_lost_byte_count: AtomicU64::new(0),
+                buffer_ready_event: Some(EventFd::new(libc::EFD_NONBLOCK).unwrap()),
```

Comment:
I would expect the `Default` for an Option<something> to be None. Can we initialize it with `Some()` only in the test that is checking the `in_buffer_empty()`? This way you can keep the `#[derive(Default)]` above as well.